documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,237 @@
+
+## Handling of acr_values
+
+OIDC specifies in [Mandatory to Implement Features for All OpenID Providers](https://openid.net/specs/openid-connect-core-1_0.html#ServerMTI) that OpenId Providers MUST implement support for acr_values.
+
+> Authentication Context Class Reference
+>>    OPs MUST support requests for specific Authentication Context Class Reference values via the acr_values parameter, as defined in Section 3.1.2. (Note that the minimum level of support required for this parameter is simply to have its use not result in an error.)
+
+OIDC also defines that the parameter acr_values is OPTIONAL and does not specify possible values. This leads to interoperability issues.
+
+This documents defines that Camara OpenId Providers MUST ignore the parameter acr_values. 
+
+This document defines that Camara Clients SHOULD not use the acr_values parameter. 
+
+> To foster interoperability a future version of this document might define values for the acr_values parameter acceptable in Camara.
+
+## Access Token Request
+
+This CAMARA document makes the scope parameter REQUIRED for the OAuth2 Client Credentials Grant.
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## The Scope Parameter
+
+
+Scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services. The scope values must be documented in the API OAS files by all Camara API subprojects. This document does not change OIDC definitions of scope values.
+
+
+---
+**NOTE**
+
+Scope values are an integral part of any OAuth2 and OIDC implementation. The RS enforces API access based on scope (if the Camara API subproject defines scopes).
+Therefore scopes should be available to API implementations.
+
+---
+
+
+## Missing "openid" scope
+
+[OIDC Core Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) states the following about the value of scope.
+> REQUIRED. OpenID Connect requests MUST contain the openid scope value. If the openid scope value is not present, the behavior is entirely unspecified. Other scope values MAY be present. ...
+
+This document defines the following error handling for a missing "openid" value in scope. 
+Please refer to [Authentication Error Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthError).
+
+If "openid" is missing in the scope value, the Authorization Server returns an HTTP response code of 400 (Bad Request) and an error invalid_request.
```

Comment:
Regarding the behavior of missing the `openid` scope, returning an `invalid_request` error can indeed impact on implementations where both OAuth2 and OIDC solutions coexist. Perhaps not returning the `id_token` ensures better interoperability.

Reply:
We are not saying not to use OIDC; we are saying that you cannot force an `invalid_request` error to be returned if `openid` is not sent when the standard states that the behavior is _entirely unspecified_, even though it is a required parameter value. If we were to return `invalid_request`, it would impact on authservers where both OIDC and OAuth2 solutions coexist. 

As a client, you SHOULD send the `openid` scope. However, if you do not send it, it is your responsibility because the behavior may vary across already deployed implementations.